### PR TITLE
Added Ugly number in c and c++

### LINKED ISSUE #896: Ugly number in c and c++

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #896 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Ugly number in c and c++

#### Languages Used:
- C
- C++

#### Files Added:
- Ugly_Number.c
- Ugly_Number.cpp

### CODE REVIEW

File: Ugly_Number/Ugly_Number.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
+//Ugly numbers are numbers whose only prime factors are 2, 3 or 5.
+
+# include <stdio.h>
+
+int main()
+{
+    int num,flag=0;
+
```
Comment (sridharjajoo) [inline Ugly_Number/Ugly_Number.c]: Can use a bool for flag

Comment (somya-kapoor) [reply Ugly_Number/Ugly_Number.c]: @sridharjajoo Is it compulsory to use bool?

Comment (sridharjajoo) [reply Ugly_Number/Ugly_Number.c]: it is a good practice as bool is specifically used for these situations

**Comment by gupta2140** [comment]
You missed a corner case here. It will output "Ugly number" when the input is 1. But it's definitely wrong. Kindly rectify it.

**Comment by somya-kapoor** [comment]
@gupta2140 But 1 is also an ugly number.

File: Ugly_Number/Ugly_Number.c (hunk 1 above)
Comment (somya-kapoor) [reply Ugly_Number/Ugly_Number.c]: @sridharjajoo I have done the required changes. Please review.

**Comment by jainaman224** [comment]
Let's wait for @gupta2140 for review.

**Comment by gupta2140** [comment]
@somya-kapoor yes, you are correct that 1 is an ugly number. I misunderstood the "ugly numbers" definition in your code. Can you rephrase it as, "An ugly number is a number whose prime factorization, if exists, has no other number other than 2, 3, 5.". Cause 1 is neither a prime number nor a composite number, and its prime factor doesn't has any of those 3 numbers. This is a special case. So, writing the comment is important. And everything else if good.

**Comment by jainaman224** [comment]
Please squash commit and it is ready to be merged.

**Comment by somya-kapoor** [comment]
@jainaman224 I have squashed the commits. Please review.
